netket/sampler/jax/metropolis_exchange.py

```diff
@@ -0,0 +1,99 @@
+import jax
+from .metropolis_hastings import MetropolisHastings
+import numpy
+
+
+class _exchange_kernel:
+    def __init__(self, hilbert, d_max):
+        clusters = []
+        distances = jax.numpy.asarray(hilbert.graph.distances())
+        size = distances.shape[0]
+        for i in range(size):
+            for j in range(i + 1, size):
+                if distances[i][j] <= d_max:
+                    clusters.append((i, j))
+
+        self.clusters = numpy.empty((len(clusters), 2), dtype=numpy.int64)
+
+        for i, cluster in enumerate(clusters):
+            self.clusters[i] = numpy.asarray(cluster)
+        self.clusters = jax.numpy.asarray(self.clusters)
```

Comment:
Just for my own understanding: Is there a reason the `numpy` and `jax.numpy` calls are mixed like this here, instead of just doing everything through `jax.numpy`?

Reply:
the only reason here was that assigning to jax arrays has a bit of boilerplate, and the syntax is cleaner with pure numpy... so basically since this is just done once in the init, I took the liberty of mixing the two!

Reply:
Okay, I see.